nextroute/model_sanity_check.go

```diff
@@ -0,0 +1,163 @@
+package nextroute
+
+import (
+	"time"
+
+	"github.com/nextmv-io/sdk/connect"
+)
+
+// SanityCheckDepth is the depth of the sanity check.
+type SanityCheckDepth int
+
+const (
+	// SanityCheckLow checks if there is at least one move per plan unit.
+	SanityCheckLow SanityCheckDepth = iota
+	// SanityCheckMedium checks same as SanityCheckLow.
+	SanityCheckMedium
+	// SanityCheckHigh count for each plan unit if it fits on a vehicle and
+	// therefor how many units fit on a vehicle.
+	SanityCheckHigh
+)
+
+// ModelSanityCheck is the interface for the sanity check.
+type ModelSanityCheck interface {
+	// Check runs the sanity check. It returns the sanity check output and an
+	// error if the sanity check itself fails. Duration is the maximum time
+	// allowed for the sanity check to run. If the sanity check takes longer
+	// the remark will be "timeout". Depth is the depth of the sanity check.
+	Check(duration time.Duration, depth SanityCheckDepth) (SanityCheckOutput, error)
+}
+
+// SanityCheckOutput is the output of the sanity check.
```

Comment:
Should some of the output struct fields below here be declared with a `omitempty`? Or will they always all be present and relevant?